### PR TITLE
Update DG on EditStudent implementation

### PR DESCRIPTION
**Pull request by revdrag**
- Add `Edit Student feature` under `Implementation` Section in `DeveloperGuide.md`
- Refactor `invalidPersonEduTrack.json` to `invalidStudentEduTrack.json` and update the file 
- Fix bug in one of the test case for `JsonEduTrackStorageTest.java`

### CODE REVIEW

**Review by seanpzk — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -159,6 +159,53 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+`Model#setStudent` and `Model#setStudentInClass` will be called to replace the specified `Student`. Finally, `Model#updateFilteredStudentList()` is called to reflect the changes made to the `Student`.
+
+Step 6. `CommandResult` is then initialised with the `String` containing `MESSAGE_EDIT_PERSON_SUCCESS` and `Student` which will then be returned.
+
+The following sequence diagram shows how the edit student operation works:
+<puml src="diagrams/EditStudentSequenceDiagram.puml" alt="EditStudentSequenceDiagram">
+
+<box type="info" seamless>
+
+**Note:** The lifeline for `EditStudentCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
```
Comment (seanpzk) [inline docs/DeveloperGuide.md:258]: Wow! I like how meticulous you are!

**Review by seanpzk — APPROVED** [review]
LGTM!

**Review by DonovanJJ — APPROVED** [review]
LGTM! 
Nice work 👍

**Pull request merged**